diff --git a/.gitignore b/.gitignore
index d3e328f..7ceb093 100644
--- a/.gitignore
+++ b/.gitignore
@@ -50,3 +50,4 @@ Thumbs.db
 !tests/mock_json_test/json/*.json
 alabastar.beve
 *.jsonc
+.qodo
diff --git a/include/glaze/core/context.hpp b/include/glaze/core/context.hpp
index 18ca34d..58202d4 100644
--- a/include/glaze/core/context.hpp
+++ b/include/glaze/core/context.hpp
@@ -106,7 +106,7 @@ namespace glz
 
    // Runtime context for configuration
    // We do not template the context on iterators so that it can be easily shared across buffer implementations
-   struct context final
+   struct context
    {
       error_code error{};
       std::string_view custom_error_message;
@@ -121,5 +121,5 @@ namespace glz
    };
 
    template <class T>
-   concept is_context = std::same_as<std::decay_t<T>, context>;
+   concept is_context = std::derived_from<std::decay_t<T>, context>;
 }
